app/src/test/java/org/simple/clinic/analytics/TimingCollectorTest.kt

```diff
@@ -0,0 +1,306 @@
+        .isEqualTo(operationCompletedAgainIn)
+  }
+
+  @Test
+  fun `finishing the collector must report to analytics with the sum of times of all completed stages`() {
+    val reporter = MockAnalyticsReporter()
+    Analytics.addReporter(reporter)
+
+    val stage1 = "Sub Operation 1"
+    val stage2 = "Sub Operation 2"
+    val stage3 = "Sub Operation 3"
+
+    val operation1StartedAt = Instant.now(clock)
+    timingCollector.start(stage1)
+    clock.advanceBy(Duration.ofMinutes(1L))
+    val operation2StartedAt = Instant.now(clock)
+    timingCollector.start(stage2)
+    clock.advanceBy(Duration.ofHours(2L))
+    timingCollector.start(stage3)
+
+    timingCollector.stop(stage1)
+    val operation1StoppedAt = operation1StartedAt
+        .plus(Duration
+            .ofMinutes(1L)
+            .plusHours(2L))
+
+    clock.advanceBy(Duration.ofMinutes(30L))
+    timingCollector.stop(stage2)
+    val operation2StoppedAt = operation1StartedAt.plus(
+        Duration
+            .ofMinutes(1L)
+            .plusHours(2L)
+            .plusMinutes(30L))
+    clock.advanceBy(Duration.ofMinutes(15L))
+
+    timingCollector.finish()
+
+    val expectedRunId = timingCollector.runId
+    val expected = MockAnalyticsReporter.Event(
+        name = "TimeTaken",
+        props = mapOf(
+            "operationName" to name,
+            "runId" to expectedRunId,
+            "timeTakenMillis" to Duration.between(operation1StartedAt, operation1StoppedAt)
+                .plus(Duration.between(operation2StartedAt, operation2StoppedAt))
+                .toMillis()
+        )
+    )
+
+    assertThat(reporter.receivedEvents.last()).isEqualTo(expected)
+  }
+
+  @Test
+  fun `finishing the collector more than once must not report to analytics again`() {
+    val reporter = MockAnalyticsReporter()
+    Analytics.addReporter(reporter)
+
+    val stage1 = "Sub Operation 1"
+    val stage2 = "Sub Operation 2"
+
+    timingCollector.start(stage1)
+    clock.advanceBy(Duration.ofMinutes(1L))
+    timingCollector.start(stage2)
+    clock.advanceBy(Duration.ofHours(2L))
+
+    timingCollector.stop(stage1)
+    clock.advanceBy(Duration.ofMinutes(30L))
+    timingCollector.stop(stage2)
+    clock.advanceBy(Duration.ofMinutes(15L))
+
+    timingCollector.finish()
+    timingCollector.finish()
+    timingCollector.finish()
+
+    assertThat(reporter.receivedEvents).hasSize(3)
+  }
+
+  @Test
+  fun `finishing the collector when there are no completed stages must have no effect`() {
+    val reporter = MockAnalyticsReporter()
+    Analytics.addReporter(reporter)
+
+    val stage1 = "Sub Operation 1"
+    val stage2 = "Sub Operation 2"
+
+    timingCollector.start(stage1)
+    clock.advanceBy(Duration.ofMinutes(1L))
+    timingCollector.start(stage2)
+
+    timingCollector.finish()
+
+    assertThat(reporter.receivedEvents).isEmpty()
+  }
+
+  @Test
+  fun `starting a stage after finishing the collector must have no effect`() {
```

Comment:
Does this mean that if in the same class we need to report multiple operations, we should create a new instance of `TimeCollector` each time?
I would like to understand this in person.

Reply:
Yes, let's talk about this in person.

Reply:
Discussed in person: Since the `finish()` operation has been removed, this test has been removed as well. The entire notion of "finishing" a timing tracker has been removed.